### PR TITLE
Create a ViewComponent to diplay businesshours in front end 

### PR DESCRIPTION
**Pull request by jducobu**
Implementation of #799

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs [hunk 1]
```diff
@@ -1,12 +1,18 @@
-﻿using System;
-using SimplCommerce.Module.Catalog.Models;
+﻿using SimplCommerce.Module.Catalog.Models;
 using SimplCommerce.Module.Core.Models;
 using SimplCommerce.Module.Core.Services;
+using System;
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs]: Sort using

File: src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs [hunk 2]
```diff
@@ -11,14 +11,30 @@ public CalculatedProductPrice(ICurrencyService currencyService)
             _currencyService = currencyService;
         }
 
-        public decimal Price { get; set; }
+        public decimal Price => PriceExcludingTax + TaxAmount;
 
-        public decimal? OldPrice { get; set; }
+        public decimal TaxAmount { get; set; }
+
+        public decimal? OldPrice => OldPriceExcludingTax.HasValue && OldTaxAmount.HasValue ? OldPriceExcludingTax.Value + OldTaxAmount.Value : default(decimal?);
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs]: Just a little format
```csharp
public decimal? OldPrice
    => OldPriceExcludingTax.HasValue && OldTaxAmount.HasValue
        ? OldPriceExcludingTax.Value + OldTaxAmount.Value
        : default(decimal?);
```

File: src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs [hunk 3]
```diff
@@ -11,14 +11,30 @@ public CalculatedProductPrice(ICurrencyService currencyService)
...
 
         public int PercentOfSaving { get; set; }
 
         public string PriceString => _currencyService.FormatCurrency(Price);
 
+        public string TaxAmountString => _currencyService.FormatCurrency(TaxAmount);
+
         public string OldPriceString => OldPrice.HasValue ? _currencyService.FormatCurrency(OldPrice.Value) : string.Empty;
+
+        public string OldTaxAmountString => OldPrice.HasValue ? _currencyService.FormatCurrency(OldTaxAmount.Value) : string.Empty;
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs]:
```csharp
public string OldTaxAmountString
    => OldPrice.HasValue
        ? _currencyService.FormatCurrency(OldTaxAmount.Value)
        : string.Empty;
```

File: src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs [hunk 4]
```diff
@@ -11,14 +11,30 @@ public CalculatedProductPrice(ICurrencyService currencyService)
...
+
+        public decimal PriceExcludingTax { get; set; }
+
+        public string PriceExcludingTaxString => _currencyService.FormatCurrency(PriceExcludingTax);
+
+        public decimal? OldPriceExcludingTax { get; set; }
+
+        public string OldPriceExcludingTaxString => OldPriceExcludingTax.HasValue ? _currencyService.FormatCurrency(OldPriceExcludingTax) : string.Empty;
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Catalog/Models/CalculatedProductPrice.cs]:
```csharp
public string OldPriceExcludingTaxString
    => OldPriceExcludingTax.HasValue
        ? _currencyService.FormatCurrency(OldPriceExcludingTax)
        : string.Empty;
```

File: src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs [hunk 5]
```diff
@@ -37,11 +49,13 @@ public class ProductThumbnail
 
         public double? RatingAverage { get; set; }
 
+        public long? TaxClassId { get; set; }
+
         public CalculatedProductPrice CalculatedProductPrice { get; set; }
 
-        public static ProductThumbnail FromProduct(Product product)
+        public static ProductThumbnail FromProduct(Product product, ICurrencyService currencyService)
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs]: I dislike this change, but still thinking if we can do it in a better way

**Comment by thiennn** [comment]
Thank you @jducobu . However it is a best practice to keep the PR focus, a PR should address one issue only so that it can be reviewed and make merge decision easier.

I will review and add comment for this PR tonight.

**Comment by thiennn** [comment]
Normally, we do not display the prices from ProductThumbnail, but in the ProductThumbnail.CalculatedProductPrice. So we don't need to format the Prices in ProductThumbnail.

For the change related to Tax, I can imagine what you want to archive. But I don't clearly understand why you want to do this. 

This pr is kind of big. and it address multiple issues. Could you please split it out, each pr focus on one issue so that we can easily review and merge

**Comment by jducobu** [comment]
I split all the PR in separate PR

**Comment by hishamco** [comment]
Hope you take my comments under consideration

File: src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs (hunk 5 above)
Comment (jducobu) [reply src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/ViewModels/ProductThumbnail.cs]: Copy

**Pull request closed without merging**